### PR TITLE
Add new Modal component

### PR DESCRIPTION
**Pull request by amarinov**

### CODE REVIEW

File: src/components/NewModal/Modal.story.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import { withInfo } from '@storybook/addon-info';
+
+import { GROUPS } from '../.././../.storybook/hierarchySeparators';
+import State from '../State/State';
+import Modal from './Modal';
+import Button from '../Button';
+import Card from '../Card';
+
+storiesOf(`${GROUPS.COMPONENTS}|NewModal`, module)
+  //   .addDecorator(withTests('SideNavItem'))
```
Comment (connor-baer) [inline src/components/NewModal/Modal.story.js]: This seems like a leftover :)

File: src/components/NewModal/Modal.js [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+  /**
+   * Callback fired when the backdrop is clicked.
+   */
+  onBackdropClick: PropTypes.func,
+  /**
+   * Callback fired when the component requests to be closed.
+   * The `reason` parameter can optionally be used to control the response to `onClose`.
+   *
+   * @param {object} event The event source of the callback
+   * @param {string} reason Can be:`"escapeKeyDown"`, `"backdropClick"`
```
Comment (connor-baer) [inline src/components/NewModal/Modal.js]: I like these annotations, should we make them a convention?

File: src/components/NewModal/ModalManager.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    this.hideSiblingNodes = hideSiblingNodes;
+    this.handleContainerOverflow = handleContainerOverflow;
+
+    // this.modals[modalIdx] = modal
+    this.modals = [];
+    // this.containers[containerIdx] = container
+    this.containers = [];
+    // this.data[containerIdx] = {
+    //   modals: [],
+    // }
```
Comment (connor-baer) [inline src/components/NewModal/ModalManager.js]: I suppose these are annotations and not forgotten comments? 🤔

File: src/components/NewModal/ModalManager.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        removeContainerStyle(data, container);
+      }
+
+      if (this.hideSiblingNodes) {
+        showSiblings(container, modal.mountNode);
+      }
+      this.containers.splice(containerIdx, 1);
+      this.data.splice(containerIdx, 1);
+    } else if (this.hideSiblingNodes) {
+      // Otherwise make sure the next top modal is visible to a screan reader.
```
Comment (connor-baer) [inline src/components/NewModal/ModalManager.js]: 👏 for accessibility.

**Comment by amarinov** [comment]
Closing in favor of #265

**Pull request closed without merging**